src/osd/OSD.cc

```diff
@@ -3391,6 +3392,25 @@ will start to track new ops received afterwards.";
     ceph_assert(m);
     m->dump_stats(f);
     m->put();
+  } else if (prefix == "dump_messenger") {
+    lock_guard l(osd_lock);
+    f->open_object_section("status");
+    f->open_object_section("messenger");
+    for (const auto& [name, msg] : {
+	std::pair{"client", client_messenger},
+	std::pair{"cluster", cluster_messenger},
+	std::pair{"hb back client", hb_back_client_messenger},
+	std::pair{"hb back server", hb_back_server_messenger},
+	std::pair{"hb front client", hb_front_client_messenger},
+	std::pair{"hb front server", hb_front_server_messenger},
+	std::pair{"objecter", objecter_messenger}
```

Comment:
I see now that there may be multiple messenger instances. You can still put this hook in the Messenger, just use the name associated with the Messenger.